pytorch_lightning/loggers/mlflow.py

```diff
@@ -230,7 +230,10 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
     @rank_zero_only
     def finalize(self, status: str = "FINISHED") -> None:
         super().finalize(status)
-        status = "FINISHED" if status == "success" else status
+        if status == "success":
+            status = "FINISHED"
+        elif status == "failed":
```

Comment:
an else is missing, which can lead to an undefined value for status. 
let's choose a default value, or not call set_terminated if status is neither finished nor failed.